inc/class-purge-cache.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * WP Irving Cache.
+ *
+ * @package WP_Irving
+ */
+
+namespace WP_Irving;
+
+/**
+ * Class to purge cache.
+ */
+class Purge_Cache {
+
+	/**
+	 * Class constructor.
+	 */
+	public function __construct() {
+		// Register admin page.
+		add_action( 'admin_menu', [ $this, 'register_admin' ] );
+
+		// Purging actions.
+		add_action( 'wp_insert_post', [ $this, 'on_update_post' ], 10, 2 );
+		add_action( 'transition_post_status', [ $this, 'on_post_status_transition' ], 10, 3 );
+		add_action( 'before_delete_post', [ $this, 'on_before_delete_post' ] );
+
+		add_action( 'init', [ $this, 'purge_cache_request' ] );
+	}
+
+	/**
+	 * Clear cache on post update.
+	 *
+	 * @param int     $post_id Post ID.
+	 * @param WP_Post $post    Post object.
+	 */
+	public function on_update_post( $post_id, $post ) {
+		if ( 'publish' !== $post->post_status ) {
+			return;
+		}
+
+		// Purge cache.
+		$this->fire_purge_request( $post_id );
+	}
+
+	/**
+	 * Clear cache on page transition.
+	 *
+	 * @param string  $new_status New Status.
+	 * @param string  $old_status Old Status.
+	 * @param WP_Post $post       Post object.
+	 */
+	public function on_post_status_transition( $new_status, $old_status, $post ) {
+		if ( 'publish' !== $new_status && 'publish' !== $old_status ) {
+			return;
+		}
+
+		// Purge cache.
+		$this->fire_purge_request( $post->ID );
+	}
+
+	/**
+	 * Clear on post delete.
+	 *
+	 * @param int $post_id Post ID.
+	 */
+	public function on_before_delete_post( $post_id ) {
+		$this->fire_purge_request( $post_id );
+	}
+
+	/**
+	 * Fire purge request.
+	 *
+	 * @param int    $post_id   Post ID.
+	 * @param string $permalink Permalink.
+	 */
+	protected function fire_purge_request( $post_id = 0, $permalink = '' ) {
```

Comment:
The signature of this method is a little confusing in that the `$permalink` parameter silently overrides the `$post_id` parameter. Could this function accept just the URL, without being aware of posts or any object type? When posts need to be purged, their permalink could be passed directly.